sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient.cs

```diff
@@ -0,0 +1,969 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Messaging.WebPubSub
+{
+    /// <summary>
+    /// Azure Web PubSub Service Client.
+    /// </summary>
+    public partial class WebPubSubServiceClient
+    {
+        private const string ApiVersion = "2020-10-01";
+
+        private const string JsonContent = "application/json";
+        private const string TextContent = "text/plain";
+
+        private Uri _endpoint;
+        private readonly string hub;
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private AzureKeyCredential _credential;
+
+        internal WebPubSubRestClient RestClient { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebPubSubServiceClient"/>.
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="hub">Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore.</param>
+        /// <param name="credential"></param>
+        public WebPubSubServiceClient(Uri endpoint, string hub, AzureKeyCredential credential) : this(endpoint, hub, credential, new WebPubSubServiceClientOptions())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebPubSubServiceClient"/>.
+        /// </summary>
+        /// <param name="endpoint"></param>
+        /// <param name="hub">Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore.</param>
+        /// <param name="credential"></param>
+        /// <param name="options"></param>
+        public WebPubSubServiceClient(Uri endpoint, string hub, AzureKeyCredential credential, WebPubSubServiceClientOptions options): this(
+            new ClientDiagnostics(options),
```

Comment:
You need to make sure `options` isn't null before passing it here.  Maybe just pass `options`, `credential`, and `endpoint` into your other .ctor instead to validate/build.